table/block.h

```diff
@@ -80,10 +88,23 @@ class BlockReadAmpBitmap {
 
     // Every new bit we set will bump this counter
     uint32_t new_useful_bytes = 0;
-    // Index of first bit in mask (start_offset / bytes_per_bit)
-    uint32_t start_bit = start_offset >> bytes_per_bit_pow_;
-    // Index of last bit in mask (end_offset / bytes_per_bit)
-    uint32_t end_bit = end_offset >> bytes_per_bit_pow_;
+    // Index of first bit in mask
+    uint32_t start_bit =
+      (start_offset >> bytes_per_bit_pow_ == last_bit_
+             ? last_bit_
```

Comment:
No need to clamp `start_bit` and `exclusive_end_bit` by `last_bit_`. It's impossible for them to be out of range: the max possible value of `exclusive_end_bit` is reached when `end_offset = block_size - 1` and `rnd_ = 0`; this max value is `exclusive_end_bit = ceil(block_size / 2**bytes_per_bit_pow_ )`, which is just within the bounds of the bitmap.

Just do:
```
uint32_t start_bit = (start_offset + (1 << bytes_per_bit_pow_) - rnd_ - 1) >> bytes_per_bit_pow_;
uint32_t exclusive_end_bit = (end_offset + (1 << bytes_per_bit_pow_) - rnd_) >> bytes_per_bit_pow_;
assert(exclusive_end_bit >= start_bit);
assert(exclusive_end_bit <= last_bit_ + 1);
```

and also dump the `end_bit` and `mid_bit` stuff and replace lines 107-133 with:
```
if (GetAndSet(start_bit) == 0) {
  uint32_t new_useful_bytes = (exclusive_end_bit - start_bit) << bytes_per_bit_pow_;
}
```

since the [start_bit, exclusive_end_bit) ranges can't overlap for non-overlapping [start_offset, end_offset] ranges.